### DISCUSSION TITLE
Provide message for security voter that denies access?

### DISCUSSION THREAD

**Discussion by gesof** [symfony/symfony#43775] [Q&A] [answered]
Goal: provide a message to users in the response that gives some indication as to WHY their request was denied. 

I could only find these workarounds by now: 

1. Listen to the kernel controller event, perform the access checks and then throw an explicit AccessDeniedException with a specific message if the check fails

2.  pass the RequestStack to the voter (constructor injection) then get the session and add a message in the flashBag to be displayed after that.

3. implement a SubjectWrapper that encloses the $subject and a $message then vote against this object. 

No solution seems perfect, but I think some improvements may be done for the next release of the Voter to cover these needs.

**Comment by lyrixx** [comment, marked as answer]
Hello, 

There is already an issue about that issue: https://github.com/symfony/symfony/issues/27995
